app/models/merchant.rb

```diff
@@ -2,4 +2,14 @@ class Merchant < ApplicationRecord
   validates :name, presence: true
   has_many :items
   has_many :invoices
+  has_many :invoice_items, through: :invoices
+  has_many :transactions, through: :invoices
+
+  def total_revenue
+    self.invoices.joins(:transactions, :invoice_items)
+    .merge(Transaction.successful)
+    .sum("invoice_items.quantity * invoice_items.price")
+  end
+
```

Comment:
it does the thing! thinking the argument will be (date = nil) or something like that? i think that gets overwritten if you pass through a different value when you apply the method. the date would come through the params. would love your thoughts on this!